File: src/models/node.js
class Node {
  constructor (parentNode, value = null, depth, childNodes = {}, isEndOfWord = false) {
    this.parentNode = parentNode
    this.childNodes = childNodes
    this.isEndOfWord = isEndOfWord
    this.value = value
    this.depth = depth
  }
  addChild(value) {
    let childNode = this.childNodes[value]
    if (!childNode) {
      childNode = new Node(this, value, this.depth + 1)
      this.childNodes[value] = childNode
    }
    return childNode
  }
}

module.exports = Node

File: py_src/classes/node.py

class Node:

    def __init__(self, value, depth, child_nodes = {}, is_end_of_word = False):
        self.value = value
        self.depth = depth
        self.child_nodes = child_nodes
        self.is_end_of_word = is_end_of_word

    def add_child(self, value):
        try:
            child_node = self.child_nodes[value]
        except:
            child_node = Node(value, self.depth + 1, {})
            self.child_nodes[value] = child_node
        return child_node

    def print_node(self, tab = '  '):
        print tab, self.value, '---', self.depth, '---', self.is_end_of_word
        for child_node_key in self.child_nodes:
            self.child_nodes[child_node_key].print_node(tab + '  ')
